src/Features/CSharp/Portable/Completion/CompletionProviders/DeclarationNameCompletionProvider.DeclarationInfo.cs

```diff
@@ -26,22 +26,25 @@ internal struct NameDeclarationInfo
                 Accessibility accessibility,
                 DeclarationModifiers declarationModifiers,
                 ITypeSymbol type,
-                IAliasSymbol alias)
+                IAliasSymbol alias = null,
+                string parameterName = null)
             {
                 PossibleSymbolKinds = possibleSymbolKinds;
                 DeclaredAccessibility = accessibility;
                 Modifiers = declarationModifiers;
                 Type = type;
                 Alias = alias;
+                ParameterName = parameterName;
             }
 
             public ImmutableArray<SymbolKind> PossibleSymbolKinds { get; }
             public DeclarationModifiers Modifiers { get; }
             public ITypeSymbol Type { get; }
             public IAliasSymbol Alias { get; }
+            public string ParameterName { get; }
             public Accessibility DeclaredAccessibility { get; }
 
-            internal static async Task<NameDeclarationInfo> GetDeclarationInfo(Document document, int position, CancellationToken cancellationToken)
+            internal static async Task<ImmutableArray<NameDeclarationInfo>> GetDeclarationInfo(Document document, int position, CancellationToken cancellationToken)
```

Comment:
Why did this need to be an array - it seems like you could have just set `ParameterName` on the existing `NameDeclarationInfo`.

Reply:
There could be multiple valid overloads if we're doing out var. The type inferrer returns this info so I thought it was worth plumbing all the way through.